### ISSUE TITLE
Allow user defined `autocomplete` values for fields

### ISSUE THREAD

**Issue by Moro-Code** [cds-snc/platform-forms-client#626]
The value for `autocomplete` is an important one for screen readers and other assistive technologies to determine the context of a field. Currently for textfields that are not email's or phone numbers we have `autocomplete="off"`. This breaks the following WCAG 2.1 AA standard https://www.w3.org/WAI/WCAG21/Understanding/identify-input-purpose.html.

Acceptance Criteria:
- [x] Allow `autocomplete` attributes to be defined for textfields through the form JSON

**Comment by jeberhardt** [comment]
@Moro-Code - I think that using the `type` for the `autoComplete` value should be removed, and that for any `email` or `phone` fields, the `autoComplete` value should be set explicitly. This will mean going into existing forms and adding the `autoComplete` values.

The primary reason for this is because while `phone` may be a valid value for both `type` and `autoComplete`, they can refer to different phone numbers, such as home and work.
